### PR TITLE
Add warnings to piaware-config for network and unsupported changes

### PR DESCRIPTION
**Pull request by eric1tran**
This will prompt user to use the piaware-restart-network command (added in piaware-support) whenever a network configuration change has been made. This avoids having to reboot to apply those changes.

This also provides warning to user when he or she attempts to change a setting that is not supported on their install (i.e. package installs attemping to change a setting only available on SD card)

### CODE REVIEW

File: programs/piaware-config/helpers.tcl [hunk 1]
```diff
@@ -66,6 +66,9 @@ proc update_config_values {argv} {
 				puts stderr "Cleared setting for $key in [$result origin $key]"
 			} else {
 				puts stderr "Set $key to $displayVal in [$result origin $key]"
+				if {[::fa_piaware_config::is_network_setting $key]} {
```
Comment (mutability) [inline programs/piaware-config/helpers.tcl:69]: Let's check the `image-type` config setting here and only show the hint if this is a sdcard image

**Review by mutability — CHANGES_REQUESTED** [review]

File: programs/piaware-config/helpers.tcl (hunk 1 above)
Comment (mutability) [reply programs/piaware-config/helpers.tcl:69]: (maybe this is an opportunity to implement something like `is_network_setting` but for all sdcard settings, and warn about configuring options that are sdcard-only on a non-sdcard install)

Comment (eric1tran) [reply programs/piaware-config/helpers.tcl:69]: Sure, let me go through the settings and determine which are SD card/non SD card

Comment (eric1tran) [reply programs/piaware-config/helpers.tcl:69]: I attempted to make use of the `-type` arg for this, (maybe supporting multiple types) but decided to go this simpler route. Let me look at this option again for SD card vs non-SD card settings

**Comment by eric1tran** [comment]
I still need to test on package installs, but I decided to use the metadata instance and add `-sdonly` and `-network` flags to determine the option type rather than maintaining another list of options. Let me know what you think.

I went through the settings on https://flightaware.com/adsb/piaware/advanced_configuration to see which are only supported on SD cards but can you check if I missed any?

Also updated the outdated README

File: package/fa_piaware_config.tcl [hunk 2]
```diff
@@ -985,45 +1009,45 @@ namespace eval ::fa_piaware_config {
...
-			{"rfkill"                -type boolean -default no}
+			{"rfkill"                -type boolean -default no -sdonly 1}
 			{"receiver-type"         -type receiver -default rtlsdr}
-			{"rtlsdr-device-index"   -default 0}
-			{"rtlsdr-ppm"            -type integer -default 0}
-			{"rtlsdr-gain"           -type gain -default max}
+			{"rtlsdr-device-index"   -default 0 -sdonly 1}
+			{"rtlsdr-ppm"            -type integer -default 0 -sdonly 1}
+			{"rtlsdr-gain"           -type gain -default max -sdonly 1}
 			{"beast-baudrate"        -type integer}
```
Comment (mutability) [inline package/fa_piaware_config.tcl:1046]: beast-baudrate is also sdcard-only

Comment (mutability) [reply package/fa_piaware_config.tcl:1046]: also all the uat-sdr-* settings

Comment (mutability) [reply package/fa_piaware_config.tcl:1046]: also network-config-style, though I highly doubt that anyone is gonna try to set that one..

Comment (eric1tran) [reply package/fa_piaware_config.tcl]: Thanks, updated.

**Review by mutability — COMMENTED** [review]
Some tab vs space cleanups needed, otherwise looks good!

**Review by mutability — APPROVED** [review]

**Comment by eric1tran** [comment]
> Some tab vs space cleanups needed, otherwise looks good!

Cleaned up. Thank you!

**Pull request merged**